bluemira/power_cycle/loads.py

```diff
@@ -0,0 +1,1065 @@
+                    class. Only non-negative integers are accepted.
+                    """,
+            ),
+            "sanity": PowerCycleError(
+                "Value",
+                f"""
+                    The attributes `load` and `model` of an instance of
+                    the {class_name} class must have the same length.
+                    """,
+            ),
+            "time": PowerCycleError(
+                "Type",
+                f"""
+                    The `time` input used to create a curve with an
+                    instance of the {class_name} class must be numeric
+                    or a list of numeric values.
+                    """,
+            ),
+        }
+        return e
+
+    # ------------------------------------------------------------------
+    # CONSTRUCTOR
+    # ------------------------------------------------------------------
+    def __init__(self, name, data_set, model):
+
+        # Call superclass constructor
+        super().__init__(name)
+
+        # Validate inputs
+        self.data_set = self._validate_data_set(data_set)
+        self.model = self._validate_model(model)
+
+        # Validate created instance
+        self._sanity()
+
+    @classmethod
+    def _validate_data_set(cls, data_set):
+        """
+        Validate 'data_set' input to be a list of `PowerData` instances.
+        """
+        data_set = super()._validate_list(data_set)
+        for element in data_set:
+            PowerData._validate(element)
+        return data_set
+
+    @classmethod
+    def _validate_model(cls, model):
+        """
+        Validate 'model' input to be a list of valid models options.
+        """
+        model = super()._validate_list(model)
+        for element in model:
+            if element not in cls._valid_models:
+                cls._issue_error("model")
+        return model
+
+    def _sanity(self):
+        """
+        Validate instance to have `data_set` and `model` attributes of
+        same length.
+        """
+        if not len(self.data_set) == len(self.model):
+            self._issue_error("sanity")
+
+    # ------------------------------------------------------------------
+    # OPERATIONS
+    # ------------------------------------------------------------------
+
+    def __add__(self, other):
+        """
+        Addition of `PowerLoad` instances is a new `PowerLoad` instance
+        with joined `load` and `model` attributes.
+        """
+
+        # Retrieve `load` attributes
+        this_set = self.data_set
+        other_set = other.data_set
+
+        # Retrieve `model` attributes
+        this_model = self.model
+        other_model = other.model
+
+        # Create and output `another`
+        another_set = this_set + other_set
+        another_model = this_model + other_model
+        another_name = "Resulting PowerLoad"
+        another = PowerLoad(another_name, another_set, another_model)
```

Comment:
PowerLoad will not be directly used by the users. What I did in my MIRA implementations was to have a Power(Load?)Manager class that adds them, depending on inputs that characterize all power loads.

The idea is to have a pre-allocated PowerLoad instance (e.g. "coil loads") with empty vectors, and when specifying that it is the sum of other instances (e.g. "coil 1 load" and "coil 2 load") the command to re-write it only takes their relevant attributes, not their names. As in:
```
all_coils = PowerLoad("All Coils",Null,Null)
coil_1 = PowerLoad("Coil 1",coil_1_set,coil_model)
coil_2 = PowerLoad("Coil 2",coil_2_set,coil_model)

all_coils = coil_1 + coil_2
```
The last line does not overwrite `all_coils`, but only takes the 'set' and 'model' parameters of the (temporary) instance created by the sum (the `another` object, with `name` "Resulting PowerLoad").

Let me know what you think.

Reply:
It's not clear to me how that would work. 

The final line will discard the `PowerLoad` object you defined on the first line. Variables are just pointers in Python. After the first line `all_coils` points to the `PowerLoad` instance you've created, but after the last line, `all_coils` now points at the `PowerLoad` instance you created in the `__add__` function. So the name will be `'Resulting PowerLoad'`, not `'All Coils'`.

Or am I misunderstanding you?

Reply:
I am sorry, I was not clear in my last comment. The code I wrote will definitely overwrite `all_coils`.

What I meant was that the action of creating and attributing values to PowerLoads is done by the PowerLoadManager class. It will sum power loads (which creates an instance of name "Resulting..."), but extract its `set` and `mod` arguments to apply them to a pre-defined PowerLoad instance that was initially created with some default/null arguments.
This way all PowerLoad instances are already defined when the PowerLoadManager class starts computing things, including the loads that will be sum of others. I believe this is preferable because I create all power loads when reading from an input file, and only compute them at a later stage.

Something like this, but applied using some method in the class (in my previous comment I meant to override the `setter` for power loads in the manager class):
```
all_coils = PowerLoad(name = "All Coils", set = Null, mod = Null)
coil_1 = PowerLoad("Coil 1",coil_1_set,coil_model)
coil_2 = PowerLoad("Coil 2",coil_2_set,coil_model)

temp_powerload = coil_1 + coil_2
all_coils.set = temp_powerload.set
all_coils.mod = temp_powerload.mod
```